### ISSUE TITLE
Update Metanorma term model with Glossarist's concept model

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/metanorma-standoc#255]
This Glossarist concept model fully addresses all ISO-IEC usage of terminology according to ISO 10241-1, and also handles terms from IEC Electropedia.

We wish to update Metanorma's term model to this new one:
https://github.com/glossarist/concept-model

This issue supersedes:
* #248 
* #249

**Comment by opoudjis** [comment]
I've updated the concept model under https://github.com/glossarist/concept-model. I won't have time to implement it before starting on IHO, but this is an initial serialisation.

The current Metanorma XML grammar for terms is:

````
term =
  element term {
    attribute id { xsd:ID }?,
    preferred+, admitted*, related*, deprecates*,
    termdomain?, termgrammar*, definition, termnote*, termexample*, termsource*
  }

preferred =
  element preferred { TextElement+ }

admitted =
  element admitted { TextElement+ }

related =
  element related {
    attribute type { "compare" | "contrast" | "see" }?,
    TextElement+
}

deprecates =
  element deprecates { TextElement+ }

termdomain =
  element domain { TextElement+ }

termgrammar =
  element grammar { TextElement+ }

definition =
  element definition {
    ( paragraph | figure | formula )+
  }

termnote =
  element termnote {
    attribute id { xsd:ID },
paragraph }

termexample =
  element termexample {
    attribute id { xsd:ID },
paragraph }

termsource =
  element termsource {
    attribute status { ( "identical" | "modified" ) },
    origin, modification?
}

origin =
  element origin { erefType | termref }

modification =
  element modification { paragraph }

termref = element termref {
  attribute base { text },
  attribute target { text },
  text?
}
````

The future state will be:

````
term =
  element term {
    attribute id { xsd:ID }?,
    preferred+, admitted*, deprecates*,
    termdomain?, termsubject?, usageinfo?, 
    defnAndOrNonverbalrep, termnote*, termexample*, related*, termsource*
  }

preferred =
  element preferred { Designation }

admitted =
  element admitted { Designation }

deprecates =
  element deprecates { Designation }

termdomain =
  element domain { TextElement+ }

termsubject =
  element subject { TextElement+ }

termsubject =
  element subject { TextElement+ }

usageinfo =
  element grammar { TextElement+ }

defnAndOrNonverbalrep =
   definition | nonverbalrep | definition, nonverbalrep

definition =
  element definition { GlossaristDefinition  }

GlossaristDefinition =
  paragraph+, termsource*

nonverbalrep = 
  element nonverbalrep {
    (  figure | formula | table )+,
    termsource*
}

termnote =
  element termnote {
    attribute id { xsd:ID },
GlossaristDefinition }

termexample =
  element termexample {
    attribute id { xsd:ID },
GlossaristDefinition }

related =
  element related {
    attribute type { RelatedConceptType },
    preferred,
    termref
}

RelatedConceptType = "deprecates" | "supersedes" | "narrower" | "broader" | "equivalent" | "compare" | "contrast" | "see"

termsource =
  element termsource {
    attribute status { ConceptSourceStatus },
    attribute type { ConceptSourceType },
    origin, modification?
}

ConceptSourceStatus = "identical" | "modified" | "restyled" | "context-added" | "generalisation" | "specialisation" | "unspecified"

ConceptSourceType = "authoritative" | "lineage"


origin =
  element origin { erefType | termref }

modification =
  element modification { paragraph }

termref = element termref {
  attribute base { text },
  attribute target { text },
  text?
}

Designation =
  attribute absent { xs:bool }?,
  attribute geographicArea { iso3166 }?,
  ( graphical_symbol_designation | letter_symbol_designation | expression_designation ),
  termsource*

graphical_symbol_designation = element graphical-symbol {
  attribute isInternational {xs:bool }?,
  figure?,
  element comment { text }?
} 

letter_symbol_designation = element letter-symbol {
   attribute isInternational {xs:bool }?,
   LocalizedString
}

expression_designation = element expression {
  attribute language { iso639 }?,
  attribute scripts { iso639 }?,
  attribute type { ExpressionType }?,
  element name { (PureTextElement | emptySlot )+ },
  element abbreviationType { AbbreviationType }?
  element pronunciation { LocalizedString }?,
  grammar?
}

ExpressionType = "prefix" | "suffix" | "abbreviation" | "full"

AbbreviationType = "truncation" | "acronym" | "initialism"

emptySlot = element emptySlot {}

grammar = element grammarInfo {
  element gender { GrammarGender }*,
  element isPreposition { xs:bool }?,
  element isParticiple { xs:bool }?,
  element isAdjective { xs:bool }?,
  element isVerb { xs:bool }?,
  element isAdverb { xs:bool }?,
  element isNoun { xs:bool }?,
  element grammarValue { text }*
}

GrammarGender = "masculine" | "feminine" | "neuter" | "common"
````

**Comment by ronaldtse** [comment]
When we deal with the term grammar refactor, we also need to consider MN AsciiDoc syntaxes that channel semantics to the underlying grammar.

For example, a synonym should be `synonym:[...]`, etc.

I guess a meta-programming approach by generating code via the grammar is possible, but not sure if it's feasible.

**Comment by opoudjis** [comment]
Model superseded by https://github.com/glossarist/concept-model

**Comment by ronaldtse** [comment]
@opoudjis just FYI, it's the same model in the original post...

**Comment by ronaldtse** [comment]
As @opoudjis rightfully pointed out, a definition is paired with notes and examples too, so the current model of only allowing "definitions + sources" is simplistic. I don't think we've run into a real-world need on this yet, but good to keep in mind.

**Comment by opoudjis** [comment]
Adding to Metanorma syntax, for https://github.com/metanorma/metanorma-ogc/issues/232

````
... defnAndOrNonverbalrep+ ...


defnAndOrNonverbalrep =
   definition { IGNORE: | nonverbalrep | definition, nonverbalrep }

definition =
  element definition { GlossaristDefinition  }

GlossaristDefinition =
  paragraph+, termsource*
````

**Comment by opoudjis** [comment]
Hitherto, term definitions have been the first paragraph encountered in a term clause, that did not involve alt or deprecated macros: it's the first proper paragraph in a term section.

We will keep that as a default, but in order to allow flexible markup of complex term structures, we will also be using open blocks with roles for the more complex structures foreseen here. In this instance, we will allow a `[.definition]` open block. It can contain paragraphs and term sources; down the road, it can also contain nonverbalrep (figures and formulas), and if it comes to that, it can also contain notes and examples if they come up.

The *rendering* of multiple definitions will (by default) be as a numbered list; I would rather not mark it up as a numbered list though, because I want to keep the composite structure as explicit as possible, rather than get into a whole lot of list continuations. (Especially if we end up automatically generating Metanorma Asciidoc out of glossarist.)

**Comment by opoudjis** [comment]
I'm not going to pass the syntactic changes along to Presentation XML: they will continue to look like they did to @Intelligent2013

**Comment by opoudjis** [comment]
Syntax change: `preferred`, `admitted`, `deprecates` from `TextElement+` to:

`Designation`:

````
Designation = expression_designation

expression_designation = element expression {
element name { PureTextElement+ }
}
````
